<launch>
        <arg name="marker_size" default="19" />
        <arg name="max_new_marker_error" default="0.08" />
        <arg name="max_track_error" default="0.2" />
        <arg name="cam_image_topic" default="/usb_cam/image_raw" />
        <arg name="cam_info_topic" default="/usb_cam/camera_info" />
        <arg name="output_frame" default="/usb_cam" />

        <node name="ar_track_alvar" pkg="ar_track_alvar" type="individualMarkersNoKinect" respawn="false" output="screen">
                <param name="marker_size"           type="double" value="$(arg marker_size)" />
                <param name="max_new_marker_error"  type="double" value="$(arg max_new_marker_error)" />
                <param name="max_track_error"       type="double" value="$(arg max_track_error)" />
                <param name="output_frame"          type="string" value="$(arg output_frame)" />

                <remap from="camera_image"  to="$(arg cam_image_topic)" />
                <remap from="camera_info"   to="$(arg cam_info_topic)" />
        </node>
	<node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen" >
		<param name="video_device" value="/dev/video0" />
		<param name="image_width" value="1280" />
		<param name="image_height" value="720" />
		<param name="pixel_format" value="yuyv" />
		<param name="camera_frame_id" value="usb_cam" />
		<param name="framerate" value="5" />
		<param name="io_method" value="mmap" />
	</node>
	<!--node pkg="tf" type="static_transform_publisher" name="link1_broadcaster" args="0.0 0.0 0.0 0.0 0.0 0.0 ar_marker_8 map 100" /-->

	<node cwd="node" name="lli_node" pkg="aauship_control" type="lli-node.py" output="screen"/>

	<node cwd="node" name="sensor_node" pkg="aauship_control" type="sensor-node.py" output="screen"/>

    <node name="pid_x" pkg="pid" type="controller" output="screen" >
      <param name="Kp" value="5.0" />
      <param name="Ki" value="0.0" />
      <param name="Kd" value="10" />
      <param name="upper_limit" value="10" />
      <param name="lower_limit" value="-10" />
      <param name="windup_limit" value="10" />
      <param name="cutoff_frequency" value="1" />
      <param name="diagnostic_period" value="0.25" />
      <param name="max_loop_frequency" value="105.0" />
      <param name="min_loop_frequency" value="1.0" />
      <remap from="setpoint" to="/pid_x/setpoint" />
      <remap from="control_effort" to="/pid_x/control_effort" />
      <remap from="state" to="/pid_x/state" />
     </node>
    <node name="pid_y" pkg="pid" type="controller" output="screen" >
      <param name="Kp" value="3.0" />
      <param name="Ki" value="0.0" />
      <param name="Kd" value="5" />
      <param name="upper_limit" value="100" />
      <param name="lower_limit" value="-100" />
      <param name="windup_limit" value="10" />
      <param name="cutoff_frequency" value="1" />
      <param name="diagnostic_period" value="0.25" />
      <param name="max_loop_frequency" value="105.0" />
      <param name="min_loop_frequency" value="1.0" />
      <remap from="setpoint" to="/pid_y/setpoint" />
      <remap from="control_effort" to="/pid_y/control_effort" />
      <remap from="state" to="/pid_y/state" />
     </node>
    <node name="pid_yaw" pkg="pid" type="controller" output="screen" >
      <param name="Kp" value="0.3" />
      <param name="Ki" value="0.0" />
      <param name="Kd" value="1.75" />
      <param name="upper_limit" value="10" />
      <param name="lower_limit" value="-10" />
      <param name="windup_limit" value="10" />
      <param name="cutoff_frequency" value="1" />
      <param name="diagnostic_period" value="0.25" />
      <param name="max_loop_frequency" value="105.0" />
      <param name="min_loop_frequency" value="1.0" />
      <remap from="setpoint" to="/pid_yaw/setpoint" />
      <remap from="control_effort" to="/pid_yaw/control_effort" />
      <remap from="state" to="/pid_yaw/state" />
     </node>


</launch>




